src/main/java/io/confluent/idesidecar/restapi/connections/ConnectionState.java

```diff
@@ -99,24 +110,54 @@ protected ConnectionStatus getInitialStatus() {
    * update the {@link #getStatus() cached results}.
    *
    * <p>This method always calls {@link #doRefreshStatus()} and then on success updates
-   * the {@link #getStatus() cached connection status}.
+   * the {@link #getStatus() cached connection status}. If we see that the status has been updated
+   * since the refresh was started, we ignore the update.
    *
    * @return the future that will complete with the updated connection status
    * @see #getConnectionStatus()
    * @see #doRefreshConnectionStatus()
    */
   public final Future<ConnectionStatus> refreshStatus() {
     var originalState = this.cachedStatus.get();
+    var beforeStartingRefresh = Instant.now();
 
     // Always set the cached status when the future completes successfully
-    return doRefreshStatus().onSuccess(updated ->
-        updateStatus(originalState, updated)
-    );
+    return doRefreshStatus().onSuccess(updated -> {
+      var lastUpdatedInstant = lastUpdated.get();
+      if (lastUpdatedInstant.isAfter(beforeStartingRefresh)) {
+        Log.infof(
+            "Ignoring stale connection status update for %s: %s. " +
+                "last updated: %s, before starting refresh time: %s",
+            spec.id(),
+            updated,
+            lastUpdatedInstant,
+            beforeStartingRefresh
+        );
+      } else {
+        // Update the status if it has not been updated since the refresh was started
+        updateStatus(originalState, updated);
+        Log.infof(
+            "Updated connection status for %s: %s, " +
+                "last updated: %s, before starting refresh time: %s",
+            spec.id(),
+            updated,
+            lastUpdatedInstant,
+            beforeStartingRefresh
+        );
+      }
+    });
   }
 
   private void updateStatus(ConnectionStatus original, ConnectionStatus updated) {
     // update the cached status
     this.cachedStatus.set(updated);
```

Comment:
Updating two 'atomic' references serially w/o an overarching lock still smells race-condition-y. An observer could observe the updated cachedStatus w/o the updated lastUpdated, couldn't they?

Reply:
Hm, I'm not sure if there was ever a "race" at all, actually. This change makes it such that refreshes started before an update don't go through. 

Currently, the only observer reading the `lastUpdated` atomic value is the refresh task. And there is always only a single instance of this per connection type. Hence, there is no possibility for multiple observers to race and read stale `lastUpdated` values.

Reply:
Added some details regarding the before/after state to the PR description.